dist/formatters/no-internal-summary-table-creator.js

```diff
@@ -0,0 +1,125 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+const path = require('path');
+const fs = require('fs');
+
+module.exports = function(messages, ruleId) {
+  const problemFiles = new Map();
+  const errorTracker = new Map();
+  const tagViolationsTracker = new Map();
+  const currentWorkingDirectory = process.cwd();
+  const maxCellWidth = 80;
+  // Ensure the current working directory ends with a path separator
+  const cwdWithSeparator = currentWorkingDirectory.endsWith(path.sep) ? currentWorkingDirectory : currentWorkingDirectory + path.sep;
+
+  function createAsciiTable(rows, maxCellWidth) {
+    // Calculate column widths
+    const colWidths = rows[0].map((_, i) => Math.min(Math.max(...rows.map(row => String(row[i]).length)), maxCellWidth));
+
+    // Create row separator
+    const rowSeparator = '+' + colWidths.map(width => '-'.repeat(width + 2)).join('+') + '+';
+
+    // Split cell contents into lines
+    const splitRows = rows.map(row => row.map((cell, i) => {
+        const cellStr = String(cell);
+        const lines = [];
+        for (let j = 0; j < cellStr.length; j += maxCellWidth) {
+            lines.push(cellStr.slice(j, j + maxCellWidth));
+        }
+        return lines;
+    }));
+
+    // Convert rows to strings
+    const rowStrings = splitRows.map(row => {
+        const numLines = Math.max(...row.map(cellLines => cellLines.length));
+        const lines = Array(numLines).fill().map((_, i) =>
+            '|' + row.map((cellLines, j) => ' ' + (cellLines[i] || '').padEnd(colWidths[j]) + ' ').join('|') + '|'
+        );
+        return lines.join('\n');
+    });
+
+    // Join everything together
+    return rowSeparator + '\n' + rowStrings.join('\n' + rowSeparator + '\n') + '\n' + rowSeparator;
+}
+
+  messages.forEach((message) => {
+    const errorMessage = message.text ?? message.message;
+    // regex pattern for error message for no internal violations
+    const re = new RegExp('(.*) "(.*)" is (.*).');
+    const match = errorMessage.match(re);
```

Comment:
Couple issues here. The first is the polynomial expansion because the regex isn't precise enough. It's not the biggest issue since the input isn't really user supplied (and this is all tooling).

I'm more concerned with how you're targeting the error message. Is there no other way to determine if it's the internal violation?

Reply:
there is another way to check, with ruleId or with messageId but that would need to be tested a bit more first. However, we need the text from that message regardless, to parse the kind, name, and tag and using this regex is the best method to do that.

We can explore options at a later time but we might be stuck with this implementation for now